# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1523421834 14400
#      Wed Apr 11 00:43:54 2018 -0400
# Node ID 2815f3257c9a90431dd8e2bd36600b40b442f556
# Parent  22522ea137b47f1b94015fdac2dd7228f4212a6e
# Parent  cbe53b5ea5ce2eb2444392e09be2967f73b4253b
maint: Merge stable to default.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -4748,19 +4748,21 @@ namespace octave
     if (len > 4 && file.substr (len-4, len-1) == ".oct")
       {
         if (autoload && ! fcn_name.empty ())
           nm = fcn_name;
 
         octave_function *tmpfcn
           = dyn_loader.load_oct (nm, file, relative_lookup);
 
-        tmpfcn->stash_package_name (package_name);
-
-        retval = octave_value (tmpfcn);
+        if (tmpfcn)
+          {
+            tmpfcn->stash_package_name (package_name);
+            retval = octave_value (tmpfcn);
+          }
       }
     else if (len > 4 && file.substr (len-4, len-1) == ".mex")
       {
         // Temporarily load m-file version of mex-file, if it exists,
         // to get the help-string to use.
 
         std::string doc_string;
 
